File: frontend/src/lib/grades.ts
// Espelho de backend services/grades.py:recalc_final.
// Se houve recuperação (makeup > 0), vale o maior entre regular e recuperação;
// caso contrário, vale a regular. Arredonda para 2 casas.
export function recalcFinal(regular: number, makeup: number): number {
  const value = makeup > 0 ? Math.max(regular, makeup) : regular;
  return Math.round(value * 100) / 100;
}

File: backend/app/services/grades.py
"""Regras de cálculo de notas, centralizadas para evitar divergência.

A nota final usa a regra: se houve prova de recuperação (makeup > 0), vale o
maior valor entre regular e recuperação; caso contrário, vale a regular.
"""


def recalc_final(regular: float, makeup: float) -> float:
    """Calcula a nota final a partir da regular e da recuperação."""
    if makeup > 0:
        return round(max(regular, makeup), 2)
    return round(regular, 2)
